File: 0784-letter-case-permutation/0784-letter-case-permutation.cpp
class Solution {
public:
    void backtrack(string s, vector<string>& res, int j) {
        res.push_back(s);
        for(int i=j; i<s.size(); i++) {
            if(s[i] <= 'z' && s[i] >= 'a') {
                s[i] -= 32;
                backtrack(s, res, i+1);
                s[i] += 32;
            }
            else if(s[i] <= 'Z' && s[i] >= 'A') {
                s[i] += 32;
                backtrack(s, res, i+1);
                s[i] -= 32;
            }
        }
    }
    vector<string> letterCasePermutation(string s) {
        vector<string> res;
        backtrack(s, res, 0);
        return res;
    }
};

File: 0784-letter-case-permutation/0784-letter-case-permutation.java
class Solution {
    void backtrack(char[] s, List<String> res, int j) {
        res.add(String.valueOf(s));
        for(int i=j; i<s.length; i++) {
            if(Character.isAlphabetic(s[i])) {
                if(Character.isUpperCase(s[i])) {
                    s[i] = Character.toLowerCase(s[i]);
                    backtrack(s, res, i+1);
                    s[i] = Character.toUpperCase(s[i]);
                }
                else {
                    s[i] = Character.toUpperCase(s[i]);
                    backtrack(s, res, i+1);
                    s[i] = Character.toLowerCase(s[i]);
                }
            }
        }
    }
    public List<String> letterCasePermutation(String s) {
        List<String> res = new ArrayList<>();
        backtrack(s.toCharArray(), res, 0);
        return res;
    }
}